File: pset6/caesar.py
import sys
import cs50

upper_base = 65
lower_base = 97
count_of_letter = 26

if len(sys.argv) != 2:
    print("You should give exactly 2 inputs")
    exit(1)
    
shift_number =int(sys.argv[1])
print("plaintext:  ", end="")
text_input = cs50.get_string()
text_len = len(text_input)

ciphertext = ""
for i in range(text_len):
    char_asc = ord(text_input[i])
    if(char_asc >= upper_base and char_asc <= upper_base + count_of_letter - 1):
        shift = (char_asc + shift_number - upper_base) % count_of_letter
        ciphertext += chr(upper_base + shift)
    elif(char_asc >= lower_base and char_asc <= lower_base + count_of_letter - 1):
        shift = (char_asc + shift_number - lower_base) % count_of_letter
        ciphertext += chr(lower_base + shift)
    else:
        ciphertext += text_input[i]

print("ciphertext: {}".format(ciphertext))

File: pset2/caesar/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#define UPPER_BASE 65
#define LOWER_BASE 97
#define COUNT_OF_LETTER 26

int main(int argc, string argv[])
{
    if (argc != 2)
    {
        printf("You should give exactly 2 inputs\n");
        return 1;
    }
    int shift_number = atoi(argv[1]);
    printf("plaintext:  ");
    string text_input = get_string();
    int text_len = strlen(text_input);
    
    printf("ciphertext: ");
    for(int i = 0; i < text_len; i++)
    {
        if(text_input[i] >= UPPER_BASE && text_input[i] <= UPPER_BASE + COUNT_OF_LETTER - 1)
        {
            int shift = (shift_number + text_input[i] - UPPER_BASE) % COUNT_OF_LETTER;
            printf("%c", UPPER_BASE + shift);
        }
        else if(text_input[i] >= LOWER_BASE && text_input[i] <= LOWER_BASE + COUNT_OF_LETTER - 1)
        {
            int shift = (shift_number + text_input[i] - LOWER_BASE) % COUNT_OF_LETTER;
            printf("%c", LOWER_BASE + shift);
        }
        else
        {
            printf("%c", text_input[i]);
        }
    }
    printf("\n");
}
